### ISSUE TITLE
can't run PackerSync after upgrade to 2.6.5

### ISSUE THREAD

**Issue by zalech** [AstroNvim/AstroNvim#1294]
### Question

I can't run :PackerSync after the last update.

**SOLUTION** https://github.com/AstroNvim/AstroNvim/issues/1294#issuecomment-1312481405

**Comment by sjava** [comment]
me too

**Comment by mehalter** [comment]
Dang, I was worried about this 🤔 let me test the upgrade path. This came because I figured out how to lazy load Packer, I thought that I had fixed it to handle the migration but let me investigate it today! Sorry for the upgrade issue :/

**Comment by matteo-hertel** [comment]
This happened to me when upgrading to 2.6.0, so is not necessarily 2.6.5 😄 
a temporary solution that worked for me is to revert locally `00a703c96ecbb1ae3c8483fdc9919898a214fa06` and
00a703c96ecbb1ae3c8483fdc9919898a214fa06
if you figure out a fix for this and you need people to test it, please hit me up

**Comment by mehalter** [comment]
@matteo-hertel could you try pulling the latest (v2.6.5) and then delete your `packer_compiled.lua` file? It should be like

```sh
rm ~/.local/share/nvim/packer_compiled.lua
```

And then open AstroNvim and see if it resolves it?

**Comment by juxuanu** [comment]
Doing that 
```
Failed to load core.bootstrap

.../site/pack/packer/start/impatient.nvim/lua/impatient.lua:100: Could not hash /home/icar/.local/share/nvim/packer_compiled.l
```

Now it opens default nvim. :-(

**Comment by Samisafool** [comment]
Removing ~/.local/share/nvim worked

**Comment by mehalter** [comment]
Ah could you also delete the cache folder and see if that helps? 

```sh
rm -rf ~/.cache/nvim
```

**Comment by juxuanu** [comment]
> Removing ~/.local/share/nvim worked

That worked.

**Comment by mehalter** [comment]
Yeah doing a full refresh would definitely work, was hoping to have a solution here that didn't require having to repull all the plugins and language servers that people behave installed.

**Comment by matteo-hertel** [comment]
pulled the latest (`25ef6a`)

ran 
```
rm ~/.local/share/nvim/packer_compiled.lua
rm -rf ~/.cache/nvim
```

Packer is back but after a packerSync I get an error 
```
.config/nvim/lua/user/init.lua:145: module 'telescope.actions' not found
```

running `rm -rf  ~/.local/share/nvim` and then running `PackerSync` has the same output, still an error for telescope

**Comment by Samisafool** [comment]
That's your user config conflicting

**Comment by zbroniszewski** [comment]
Is this expected to be an issue for anybody that upgraded to 2.6.5?

**Comment by mehalter** [comment]
I think so @zbroniszewski . Did you have this issue? Im Going to make an announcement about it across all the socials

**Comment by mehalter** [comment]
Just to summarize for anyone that runs into this:

This is the easiest way to resolve the new lazy loaded packer migration without having to reinstall plugins or anything:

```sh
rm ~/.local/share/nvim/packer_compiled.lua
rm -rf ~/.cache/nvim
```

To be extra safe, you can also do a full refresh:

```sh
rm ~/.local/site/nvim
rm ~/.local/share/nvim
rm -rf ~/.cache/nvim
```

After each of these, restart AstroNvim and run `:PackerSync`.


Sorry again for this inconvenience! This shouldn't happen again since we won't be moving packer again. 😞

**Comment by mehalter** [comment]
I have also gone ahead and pinned this issue so that it remains visible for people who go to the repo to ask this question. I'll make a post on Discord and r/AstroNvim just to be safe as well.

**Comment by zbroniszewski** [comment]
@mehalter I have not upgraded to 2.6.5 quite yet. Is there any way to avoid this issue for somebody who is preparing to upgrade?

**Comment by Mjb141** [comment]
@zbroniszewski Just don't remove Packer. When asked hit `n`.

**Comment by mehalter** [comment]
@zbroniszewski @Mjb141 after you say no one time it will have the recompiled packer. On the endt startup it will ask you again to remove the packer installed to the `start` plugin folder. You should say yes the second time because it will have recompiled at that point the new lazy loading. So it should work

**Comment by zbroniszewski** [comment]
Thanks @mehalter I will test this carefully and report back. If successful, perhaps we should consider another announcement for users who have not yet upgraded to 2.6.5. (Rather than allowing 2.6.5 to break things, then fix, in case there are other unknown side effects)

**Comment by mehalter** [comment]
Yeah that would be great @zbroniszewski let me know how your testing goes. Thanks so much for the help and investigation !

**Comment by keturiosakys** [comment]
Even after this refresh I can't seem to get `PackerSync` to work :( I ran the full refresh (deleting share/nvim and cache/nvim) folders but after restarting nvim I get a whole bunch of errors and Packer fails to start. Of the errors, this seems to be the most relevant one:
```
Error in packer_compiled: ~/.config/nvim/lua/packer_compiled.lua:653: Vim(packadd):E919: Directory not found in 'packpath': "pack/*/opt/mason-lspconfig.nvim"
```

**Comment by mehalter** [comment]
@keturiosakys it looks like you need to delete `~/.config/nvim/lua/packer_compiled.lua`

**Comment by keturiosakys** [comment]
Yeah I tried that as well with no dice. I did a full reinstall of AstroNvim and managed to get rid of the above error but now getting this one:

```
[...]
.config/nvim/lua/user/lsp.lua:2: module 'cmp_nvim_lsp' not found:
^Ino field package.preload['cmp_nvim_lsp']
```

**Comment by keturiosakys** [comment]
Oh hold on I got it working now.

1. Reinstalled AstroNvim.
2. Re-added my personal config to `user` folder
3. Removed the `.local/share/nvim` (that included `packer_compiled.lua`), `.cache/nvim` etc
4. Reran Nvim and then `PackerSync` command became available

**Comment by benvds** [comment]
This worked for me neither, I had to revert back to v.2.6.4

**Comment by mehalter** [comment]
@benvds can you please share more details

**Comment by alexandersix** [comment]
Hi @mehalter, this may be me not understanding fully, but I have completely removed AstroNvim from my system, removed `packer_compiled.lua`, removed `.local/share/nvim`, and re-installed AstroNvim using the directions on the website, and I am still getting the `Not an editor command: PackerSync` error.

All I've achieved by doing this is reverting entirely back to the default install of Neovim. Any ideas where to go from here?

**Comment by mehalter** [comment]
@alexandersix once you have compiled and verified that the new code is in the `packer_compiled.lua` you will want to remove the packer installation in the `pack/start` folder because our stuff installs packer to the `opt` folder and will lazy load it.

**Comment by alexandersix** [comment]
@mehalter Where can I find the `pack/start` folder? AstroNvim is working again with the defaults, but if I try to add my user config that I've been using, everything breaks with a `Failed to load core.bootstrap` error because it couldn't hash the `packer_compiled.lua` file

**Comment by mehalter** [comment]
@alexandersix that error would be happening because you didn't clear your Neovim cache. Please make sure to follow the steps I laid out above :)

**Comment by alexandersix** [comment]
@mehalter Strange, I cleared it all out and followed those steps prior to needing to re-install packer (those steps are what caused me to not be able to to run Packer until I re-installed it). Not sure why those steps didn't work the first time, but all seems to be well over here this time.

**Comment by anupj** [comment]
I followed the recommended solution steps and it worked for me. 
Thanks again for your work on Astronvim @mehalter 👍

**Comment by daniel-hampton** [comment]
I ran into the same difficult as @alexandersix after following the recommended approach from @mehalter. After reading the later message from @mehalter that I should say "No" to the first time it asks me if I want to remove packer from the 'start' folder, my second attempt worked. I was able to say yes to removing it on the second prompt after it recompiled and looks like everything is working. Thanks for the information in this thread and your work on this software. :+1:

**Comment by benvds** [comment]
i also got things working again by clearing all:
```terminal
rm -rfd ~/.config/nvim/lua/packer_compiled.lua; rm -rfd ~/.local/share/nvim ; rm -rfd ~/.cache/nvim/;
```
and then doing a clean checkout

**Comment by talksik** [comment]
same issue with all of the tips

**Comment by chhh** [comment]
```
Error detected while processing command line:
492: Not an editor command: PackerSync
```

**Comment by maxdevjs** [comment]
This project is so cool 😊😍🚀

P.S.: perhaps a well noticeable, temporary, note/walk-through to solve this (or future) kind of 😱 issues could be provided in the README 😀

**Comment by Rafael-Conde** [comment]
> Just to summarize for anyone that runs into this:
> 
> This is the easiest way to resolve the new lazy loaded packer migration without having to reinstall plugins or anything (**Preferred method**):
> 
> ```shell
> rm ~/.local/share/nvim/packer_compiled.lua
> rm -rf ~/.cache/nvim
> ```
> 
> To be extra safe, you can also do a full refresh (**Last resort**):
> 
> ```shell
> rm -rf ~/.local/site/nvim
> rm -rf ~/.local/share/nvim
> rm -rf ~/.cache/nvim
> ```
> 
> After each of these, restart AstroNvim and run `:PackerSync`.
> 
> Sorry again for this inconvenience! This shouldn't happen again since we won't be moving packer again. 😞

The windows paths are:

```
C:\Users\rafae\.cache\nvim
```

```
C:\Users\rafae\AppData\Local\nvim-data\packer_compiled.lua
```

```
C:\Users\rafae\AppData\Local\nvim-data\pack
```

One can delete them by sticking a `del `  before the path on cmd.

```
del C:\Users\rafae\.cache\nvim
del C:\Users\rafae\AppData\Local\nvim-data\packer_compiled.lua
del C:\Users\rafae\AppData\Local\nvim-data\pack
```

**Comment by mehalter** [comment]
Thanks for this @Rafael-Conde ! I'll also link it at the top!

**Comment by Rafael-Conde** [comment]
It's a pleasure to be of help @mehalter !!! Let me know if there is anything else I could do!!

**Comment by convers39** [comment]
I keep facing this issue after updating to the latest 2.9.1 even when I tried to remove all related folders and reinstall. At last I just follow the error message that I install packer manually:

```bash
git clone --depth 1 https://github.com/wbthomason/packer.nvim\
 ~/.local/share/nvim/site/pack/packer/start/packer.nvim
```

Then open AstroNvim again, and it worked. Now just enter 'y' when it asked to remove the packer installation.

**Comment by mehalter** [comment]
@convers39 no worries, you can say yes there and remove the packer installed in the start folder

**Comment by wtanksleyjr** [comment]
Hey, this is my first time trying nvim, I was recommended this ... but apparently this issue's got me anyhow, even though I don't think I'm upgrading or anything. (Maybe in a previous life I tried Packer before uninstalling nvim?)

Is there some even more extreme "kill everything" set of commands I can run (ha ha "sudo rm /") to just not deal with this at all? Given that none of the commands so far posted help at all?

**Comment by wtanksleyjr** [comment]
> Hey, this is my first time trying nvim, I was recommended this ... but apparently this issue's got me anyhow

SOLVED. It wasn't related to this problem. The problem was I was using Ubuntu's neovim, which is old. Removed that, installed from nvim's official recent deb, and BOOM everything worked.

Responding in case someone else has the same problem :) .

**Comment by coolbrd** [comment]
Still experiencing this issue. I've blown away everything relating to AstroNvim and Nvim itself and tried to reinstall per the instructions multiple times. I've tried Nvim 0.9.0 and the recommended 0.8.0, nothing. I've also successfully installed Packer on a clean install of Nvim 0.8.0, which I then tried to install AstroNvim on top of. No dice, same issue. Nothing suggested in this thread works for me, I feel like I'm going a little crazy. Is anybody else still experiencing this??
